### ISSUE TITLE
IDE0005 not reported as build warning

### ISSUE THREAD

**Issue by fabio-muramatsu** [dotnet/roslyn#58103]
I have a simple project where I'm trying to enforce code style rules in build. It seems to be mostly working, but for some reason I noticed that IDE0005 warning doesn't show up when building the project, even though it is set as warning in my editorconfig file.

Here's what my csproj looks like:
```xml
<Project Sdk="Microsoft.NET.Sdk">
  
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <AnalysisLevel>latest-Recommended</AnalysisLevel>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
  </PropertyGroup>
  
</Project>
```

Here's my [editorconfig](https://github.com/dotnet/roslyn/files/7651893/editorconfig.zip) file. I've followed the instruction from [this thread](https://github.com/dotnet/roslyn/issues/33558) and set the severities using the IDExxxx code so they can be recognized by the compiler. I'm explicitly covering IDE0005 in my editorconfig:

```
dotnet_diagnostic.IDE0005.severity = warning # Remove unnecessary import
```

And here's a Program.cs file that triggers a bunch of code style warnings:

```cs
using System;
using System.Threading.Tasks;

namespace CSSandbox
{
    class Program
    {
        static async Task Main(string[] args)
        {

        }
    }
}
```

Building this, the compiler reports many IDExxxx warnings, which indicates that my editorconfig file is being at least partially taken into account when building:

```
Program.cs(8,27,8,31): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
Program.cs(4,1,4,20): warning IDE0161: Convert to file-scoped namespace
Program.cs(8,41,8,45): warning IDE0060: Remove unused parameter 'args'
Program.cs(8,27,8,31): warning IDE0040: Accessibility modifiers required
Program.cs(6,11,6,18): warning IDE0040: Accessibility modifiers required
```

IDE0005 is missing from that list, though. I can see it being detected in the VS UI:
![image](https://user-images.githubusercontent.com/14302074/144661427-c0ee4a69-d78d-4c92-ada3-fae33cb2fa5f.png)

Is there anything I'm missing here? When I change the severity of IDE0005 in my editorconfig file, it reflects in the VS UI, but the build doesn't seem to catch it. It tried both building this from VS and from the command line, and the results are the same. I'd appreciate any pointers here.

**Comment by Youssef1313** [comment]
cc @mavasani

**Comment by mavasani** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/41640#issuecomment-985780130
